### ISSUE TITLE
[Suggestion] Create Unity Package instead

### ISSUE THREAD

**Issue by zeh** [josec89/wakatime-unity#21]
Unity is moving towards a "package" architecture. Now, features can be "installed" in projects and be managed externally, as packages, without polluting the game's code base. While this is on early days, the intention is for this to work similarly to NPM packages - that is, you can install third-party dependencies at will. Unity 2018.1 is the first one to feature UPM, the Unity Package Manager.

WakaTime support is an ideal target for that. Pasting WakaTime code into a game just so you can get stats seems kind of icky.

Here's an introduction to the changes:
https://blogs.unity3d.com/2018/05/04/project-management-is-evolving-unity-package-manager-overview/

More about UPM:
https://docs.unity3d.com/Packages/com.unity.package-manager-ui@1.8/manual/index.html
https://forum.unity.com/forums/package-manager.150/

_Currently_, only official (Unity-created) packages are allowed. However, the intention is to support packages using a Git URL in the future.

In the meantime, we can "install" packages by copying them locally and referring to them by their relative path location. They would be _out_ of the project folder.

Here's instructions on how to do that:
https://gist.github.com/LotteMakesStuff/6e02e0ea303030517a071a1c81eb016e

This should serve as a great preparation for a Git UPM package for WakaTime.

TL;DR: in the future, developers will be able to use WakaTime inside Unity by just pasting one line of code into their `manifest.json`, rather than downloading scripts and placing them inside the project. Making this available should probably be a priority for WakaTime Unity plugin maintainers.

**Comment by josec89** [comment]
That sounds great :) maybe you can place a PR with the needed changes?

**Comment by zeh** [comment]
Will take a look at that once Git support is out.

**Comment by nstringham** [comment]
@zeh is git support out?

**Comment by zeh** [comment]
> @zeh is git support out?

It is, but it's been a while. I haven't kept up to date. My suggestion is still to release it as a package, but I don't have the full context anymore to be able to implement, test, and create a PR myself (sorry).
